tests/core/nlg/test_response.py

```diff
@@ -80,6 +80,27 @@ async def test_nlg_when_multiple_conditions_satisfied():
     assert resp.get("text") in ["example A", "example B"]
 
 
+async def test_nlg_slot_case_sensitivity():
+    responses = {
+        "utter_action": [
+            {
+                "text": "example a",
+                "condition": [{"type": "slot", "name": "test", "value": "Cold"}],
+            }
+        ]
+    }
+
+    t = TemplatedNaturalLanguageGenerator(responses=responses)
+    slot_a = TextSlot(
```

Comment:
You should test with `CategoricalSlot` instance instead of `TextSlot`, where you pass the possible values via mappings argument. Then if you want to set the value of the slot, you can assign via the setter `slot_a.value = 'cold'`, rather than using `initial_value`.

Reply:
My test case fails if I don't use the `initial_value`. I think I'm making a mistake somewhere, which I'm not able to figure it out

Reply:
I'm investigating this today once I get a minute after all the morning meetings, I see that the slot value gets set after using the setter, however this doesn't seem to get passed to the dialogue state tracker 🤔

Reply:
Ok, found the culprit for why the slot value gets reset when the dialogue state tracker gets instantiated - essentially this line [here](https://github.com/RasaHQ/rasa/blob/98777d4ff91e232a10f2e29e81880c5e102db241/rasa/shared/core/trackers.py#L220). One way to avoid this is to use the class method `from_events` (e.g. `tracker = DialogueStateTracker.from_events(sender_id="test_nlg", evts=[SlotSet("test", "cold")], slots=[slot_a])`) or to keep consistent with the other tests in this module, use the initial value arg when instantiating the categorical slot, and discard the setter approach. Apologies for the confusion with the setter 🙈

Reply:
Thanks a lot for the help, It helped me to learn more about DialogueStateTracker :)